apps/mobile/src/app/wallet/wallet-settings/configure/[wallet]/view-secret-key.tsx

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useState } from 'react';
+import { ScrollView } from 'react-native-gesture-handler';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+import { Button } from '@/components/button';
+import { MnemonicDisplay } from '@/components/create-new-wallet/mnemonic-display';
+import { useToastContext } from '@/components/toast/toast-context';
+import { useCreateWallet } from '@/hooks/create-wallet';
+import { mnemonicStore } from '@/state/storage-persistors';
+import { t } from '@lingui/macro';
+import { useTheme } from '@shopify/restyle';
+import * as Clipboard from 'expo-clipboard';
+import { useLocalSearchParams } from 'expo-router';
+
+import {
+  Box,
+  CopyIcon,
+  QuestionCircleIcon,
+  Text,
+  Theme,
+  TouchableOpacity,
+} from '@leather.io/ui/native';
+
+function ViewSecretKey({ fingerprint }: { fingerprint: string }) {
+  const { bottom } = useSafeAreaInsets();
+  const theme = useTheme<Theme>();
+  const { navigateAndCreateWallet } = useCreateWallet();
+  const { displayToast } = useToastContext();
+  const [mnemonic, setMnemonic] = useState<string | null>(null);
+  useEffect(() => {
+    mnemonicStore(fingerprint)
```

Comment:
Maybe we can start adding `await` for some of these promises? I see this warning a few times